lib/resources/postcards.js

```diff
@@ -1,124 +1,78 @@
-    offset = options.offset || 0;
-  }
-
-  request(
-    {
-      url: this.uri,
-      auth: {
-        user: this.apiKey,
-        password: ''
-      },
-      qs: {
-        count: count,
-        offset: offset
-      },
-      headers: this.headers,
-      json: true,
-      method: 'GET'
-    }, function (e, r, body) {
-      if (body && body.errors) {
-        done(body.errors, body);
-      } else {
-        done(e, body);
-      }
-    });
+var fs           = require('fs');
+var util         = require('util');
+var ResourceBase = require('./resourceBase');
+
+var Postcards = function(config) {
+  ResourceBase.call(this, 'postcards', config);
 };
 
-Postcards.prototype.create = function (params, done) {
-  var opts = {
-    url: this.uri,
-    auth: {
-      user: this.apiKey,
-      password: ''
-    },
-    headers: this.headers,
-    json: true
-  };
+util.inherits(Postcards, ResourceBase);
+
+(function() {
 
-  var req = request.post(opts, function (e, r, body) {
-    if (body && body.errors) {
-      done(body.errors, body);
-    } else {
-      done(e, body);
+  this.list = function(options, callback) {
+
+    if(typeof options === 'function') {
+      callback = options;
+      options  = {};
     }
-  });
-  var form = req.form();
-
-/* istanbul ignore else */
-  if (params.front) {
-    if (!(params.front instanceof Buffer) && params.front.substr(0,1) === '@') {
-      params.front = fs.createReadStream(params.front.substr(1));
-      form.append('front', params.front);
-    } else if (params.front instanceof Buffer) {
-      form.append('front', params.front, {filename: 'front.pdf'});
-    } else {
-      form.append('front', params.front);
+
+    return this._transmit('GET', null, options, null, null, callback);
+  };
+
+  this.retrieve = function(id, callback) {
+    return this._transmit('GET', id, null, null, null, callback);
+  };
+
+  this.create = function(params, callback) {
+
+    var isBuffer;
+
+    if(params.front) {
+      isBuffer = Buffer.isBuffer(params.front);
+
+      if(isBuffer) {
+        params.front = {
+          value: params.front,
+          options: { filename: 'front.pdf' }
+        };
+      }
+
+      if(!/^http/.test(params.front) && !isBuffer) {
+        params.front = fs.createReadStream(params.front);
+      }
     }
-  }
-
-  if (params.back) {
-    if (!(params.back instanceof Buffer) && params.back.substr(0,1) === '@') {
-      params.back = fs.createReadStream(params.back.substr(1));
-      form.append('back', params.back);
-    } else if (params.back instanceof Buffer) {
-      form.append('back', params.back, {filename: 'back.pdf'});
-    } else {
-      form.append('back', params.back);
+
+    if(params.back) {
+      isBuffer = Buffer.isBuffer(params.back);
+
+      if(isBuffer) {
+        params.back = { value: params.back, options: { filename: 'back.pdf' } };
+      }
+
+      if(!/^http/.test(params.back) && !isBuffer) {
+        params.back = fs.createReadStream(params.back);
```

Comment:
The tests should be rewritten to actually pass a buffer. This was something we did incorrectly in the first version so I’d like to correct i there.

On Mon, Mar 23, 2015 at 2:11 PM, Joshua McGinnis <notifications@github.com>
wrote:

>> -      params.back = fs.createReadStream(params.back.substr(1));
>> -      form.append('back', params.back);
>> -    } else if (params.back instanceof Buffer) {
>> -      form.append('back', params.back, {filename: 'back.pdf'});
>> -    } else {
>> -      form.append('back', params.back);
>> +
>> +    if(params.back) {
>> +      isBuffer = Buffer.isBuffer(params.back);
>> +
>> +      if(isBuffer) {
>> +        params.back = { value: params.back, options: { filename: 'back.pdf' } };
>> +      }
>> +
>> +      if(!/^http/.test(params.back) && !isBuffer) {
>> +        params.back = fs.createReadStream(params.back);
> Cool. So that will break a few tests - should they be removed altogether or would you prefer to have them modified to create the buffer beforehand?
> Example: https://github.com/joshuamcginnis/lob-node/blob/promise-refactor/test/areas.js#L171-182
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/lob/lob-node/pull/115/files#r26980477